src/Middlewares/Middleware.php

```diff
@@ -69,13 +70,22 @@ public function before(Request $request)
             return;
         }
 
-        if (!$this->pluginConfig->getShopKey() || !$this->searchService->aliveTest()) {
+        if (!$this->pluginConfig->getShopKey()) {
             return;
         }
 
         $this->eventDispatcher->listen(
             'IO.Resources.Import',
             function (ResourceContainer $container) {
+                /** @var CategoryService $categoryService */
+                $categoryService = pluginApp(CategoryService::class);
+                $isCategoryPage = $categoryService->getCurrentCategory() !== null;
+                $isInSearchOrCategoryPage = $this->isSearchPage || $isCategoryPage;
```

Comment:
- We should probably initialize `$this->isSearchPage` before that, just to be sure
- We should check for `$this->activeOnCatPage` as well